python/tank_vendor/shotgun_deploy/descriptor.py

```diff
@@ -295,3 +295,63 @@ def find_latest_version(self, constraint_pattern=None):
         return latest
 
 
+###########################################################################
+# backward compatibility support - not for external use
+
+def get_legacy_cache_path(
```

Comment:
I started out that way, but this avoided circular dependency. And since it's strictly a function to preserve legacy behavior, I thought it made sense to keep it as separate as possible. Let me know if you'd prefer the other way around as I'm not tied to this. :)